python_ta/checkers/infinite_loop_checker.py

```diff
@@ -45,6 +50,159 @@ def _check_condition_all_var_used(self, node: nodes.While) -> None:
                 node=node.test,
             )
 
+    def _check_condition_constant(self, node: nodes.While) -> None:
+        """Helper function that checks if a constant while-loop condition may lead to an infinite loop.
+
+        This helper flags loops that meet **both** of the following criteria:
+            - The `while` condition is constant (e.g., `while 1 < 2`)
+            - The loop body contains no `return`, `break`, `raise`, `yield`, or `sys.exit()` calls
+        """
+        if not self._check_constant_loop_cond(
+            node, node.test
+        ) and not self._check_constant_comp_test(node):
+            return
+
+        inferred = utils.safe_infer(node.test)
+        if isinstance(inferred, util.UninferableBase) or inferred is None:
+            return
+        if isinstance(inferred, nodes.Const) and bool(inferred.value) is False:
+            return
+
+        check_nodes = (nodes.Break, nodes.Return, nodes.Raise, nodes.Yield)
+        for child in node.body:
+            for exit_node in child.nodes_of_class(klass=(nodes.Call, *check_nodes)):
+                if isinstance(exit_node, check_nodes):
+                    return
+                # Check Call node to see if `sys.exit()` is called
+                elif (
+                    isinstance(exit_node, nodes.Call)
+                    and isinstance(exit_node.func, nodes.Attribute)
+                    and exit_node.func.attrname == "exit"
+                ):
+                    inferred = utils.safe_infer(exit_node.func.expr)
+                    if (
+                        not isinstance(inferred, util.UninferableBase)
+                        and inferred is not None
+                        and isinstance(inferred, nodes.Module)
+                        and inferred.name == "sys"
+                    ):
+                        return
+        else:
+            self.add_message("infinite-loop", node=node.test, confidence=INFERENCE)
+
+    def _check_constant_comp_test(self, node: nodes.While) -> bool:
```

Comment:
Hi @Karl-Michaud, since this PR is still marked as a draft, I'm just leaving a quick review here about the new part you added.

Overall I think `_check_constant_comp_test` is good, but here are two comments:

1. I think the name (and function) is unduly narrow. This won't just check for "comparison of constant notes", but really any Python expression that involves only constants (for example, `1 + 2`). And I think that's a good thing!
2. You're right abut the duplication of inference not being ideal. I think your suggestion of just having the inference done in the "main" function is a good one.